libs/dal/src/repositories/base-repository.ts

```diff
@@ -1,10 +1,16 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import { ClassConstructor, plainToInstance } from 'class-transformer';
 import { Document, Model, Types, ProjectionType } from 'mongoose';
+import { CacheService } from '../shared';
+
 export class BaseRepository<T_Query, T_Response> {
   public _model: Model<any & Document>;
 
-  constructor(protected MongooseModel: Model<any & Document>, protected entity: ClassConstructor<T_Response>) {
+  constructor(
+    protected MongooseModel: Model<any & Document>,
+    protected entity: ClassConstructor<T_Response>,
+    protected cacheService?: CacheService
```

Comment:
shouldn't we use the interface `ICacheService` here instead of class?

Reply:
Yes, thanks for noticing that was the idea behind the creation of the interface.